# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ModuleLoadRequest.cpp
# Commit: 0a3d9301244b
# Full Hash: 0a3d9301244b91559c4c19314ce991525d9f605c
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 21:52:29
# Regressor Bug: 1742437
# File Overlap Count: 1
# Description:
#   Bug 1742437 - move mElement from ScriptFetchOptions to DOMScriptLoadContext; r=jonco,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D136742
# ==============================================================================

diff -r 46d45c7bae2f -r 0a3d9301244b dom/script/ModuleLoadRequest.cpp
--- a/dom/script/ModuleLoadRequest.cpp	Fri Feb 18 12:45:48 2022 +0000
+++ b/dom/script/ModuleLoadRequest.cpp	Fri Feb 18 12:45:49 2022 +0000
@@ -52,13 +52,13 @@
 
 /* static */
 ModuleLoadRequest* ModuleLoadRequest::CreateTopLevel(
-    nsIURI* aURI, ScriptFetchOptions* aFetchOptions,
+    nsIURI* aURI, ScriptFetchOptions* aFetchOptions, Element* aElement,
     const SRIMetadata& aIntegrity, nsIURI* aReferrer, ScriptLoader* aLoader) {
   auto* request = new ModuleLoadRequest(
       aURI, aFetchOptions, aIntegrity, aReferrer, true, /* is top level */
       false,                                            /* is dynamic import */
       aLoader->GetModuleLoader(), NewVisitedSetForTopLevelImport(aURI));
-  DOMScriptLoadContext* context = new DOMScriptLoadContext(request);
+  DOMScriptLoadContext* context = new DOMScriptLoadContext(aElement, request);
   request->mLoadContext = context;
   return request;
 }
@@ -71,7 +71,8 @@
                             aParent->mURI, false, /* is top level */
                             false,                /* is dynamic import */
                             aParent->mLoader, aParent->mVisitedSet);
-  DOMScriptLoadContext* context = new DOMScriptLoadContext(request);
+  DOMScriptLoadContext* context =
+      new DOMScriptLoadContext(aParent->mLoadContext->mElement, request);
   context->mIsInline = false;
   context->mScriptMode = aParent->GetLoadContext()->mScriptMode;
 
@@ -83,8 +84,9 @@
 /* static */
 ModuleLoadRequest* ModuleLoadRequest::CreateDynamicImport(
     nsIURI* aURI, ScriptFetchOptions* aFetchOptions, nsIURI* aBaseURL,
-    ScriptLoader* aLoader, JS::Handle<JS::Value> aReferencingPrivate,
-    JS::Handle<JSString*> aSpecifier, JS::Handle<JSObject*> aPromise) {
+    Element* aElement, ScriptLoader* aLoader,
+    JS::Handle<JS::Value> aReferencingPrivate, JS::Handle<JSString*> aSpecifier,
+    JS::Handle<JSObject*> aPromise) {
   MOZ_ASSERT(aSpecifier);
   MOZ_ASSERT(aPromise);
 
@@ -93,7 +95,7 @@
       true, /* is dynamic import */
       aLoader->GetModuleLoader(), NewVisitedSetForTopLevelImport(aURI));
 
-  DOMScriptLoadContext* context = new DOMScriptLoadContext(request);
+  DOMScriptLoadContext* context = new DOMScriptLoadContext(aElement, request);
   context->mIsInline = false;
   context->mScriptMode = DOMScriptLoadContext::ScriptMode::eAsync;
   request->mLoadContext = context;